### ISSUE TITLE
Totally different output: Chapter 9. Up and Running with TensorFlow 

### ISSUE THREAD

**Issue by achintyasarkar** [ageron/handson-ml#237]
Hi,
I am getting totally different output for the regression problem using "Implementing Gradient Descent".
Could you please let me know why it is happening (in Bash shell, ubuntu 14.04 system),
Code and output as follows,

import tensorflow as tf
import numpy as np
from sklearn.datasets import fetch_california_housing
def reset_graph(seed=42):
    tf.reset_default_graph()
    tf.set_random_seed(seed)
    np.random.seed(seed)
import os
from sklearn.preprocessing import StandardScaler

os.environ['TF_CPP_MIN_LOG_LEVEL']='2'

housing = fetch_california_housing()
m, n = housing.data.shape
housing_data_plus_bias = np.c_[np.ones((m, 1)), housing.data]

scaler = StandardScaler()
scaled_housing_data = scaler.fit_transform(housing.data)
scaled_housing_data_plus_bias = np.c_[np.ones((m, 1)), scaled_housing_data]

print(scaled_housing_data_plus_bias.mean(axis=0))
print(scaled_housing_data_plus_bias.mean(axis=1))
print(scaled_housing_data_plus_bias.mean())
print(scaled_housing_data_plus_bias.shape)

reset_graph()

n_epochs = 1000
learning_rate = 0.01
X = tf.constant(scaled_housing_data_plus_bias, dtype=tf.float32, name="X")
y = tf.constant(housing.target.reshape(-1, 1), dtype=tf.float32, name="y")
theta = tf.Variable(tf.random_uniform([n + 1, 1], -1.0, 1.0, seed=42), name="theta")
y_pred = tf.matmul(X, theta, name="predictions")
error = y_pred - y
mse = tf.reduce_mean(tf.square(error), name="mse")
gradients = 2/m * tf.matmul(tf.transpose(X), error)
training_op = tf.assign(theta, theta - learning_rate * gradients)

init = tf.global_variables_initializer()

with tf.Session() as sess:
    sess.run(init)

    for epoch in range(n_epochs):
        if epoch % 100 == 0:
            print("Epoch", epoch, "MSE =", mse.eval())
        sess.run(training_op)

    best_theta = theta.eval()
    

[  1.00000000e+00   6.60969987e-17   5.50808322e-18   6.60969987e-17
  -1.06030602e-16  -1.10161664e-17   3.44255201e-18  -1.07958431e-15
  -8.52651283e-15]
[ 0.38915536  0.36424355  0.5116157  ..., -0.06612179 -0.06360587
  0.01359031]
0.111111111111
(20640, 9)
('Epoch', 0, 'MSE =', 9.1615429)
('Epoch', 100, 'MSE =', 9.1615429)
('Epoch', 200, 'MSE =', 9.1615429)
('Epoch', 300, 'MSE =', 9.1615429)
('Epoch', 400, 'MSE =', 9.1615429)
('Epoch', 500, 'MSE =', 9.1615429)
('Epoch', 600, 'MSE =', 9.1615429)
('Epoch', 700, 'MSE =', 9.1615429)
('Epoch', 800, 'MSE =', 9.1615429)
('Epoch', 900, 'MSE =', 9.1615429)
[[-0.1673944 ]
 [-0.46283674]
 [-0.04063368]
 [-0.27085733]
 [ 0.90942287]
 [ 0.88372922]
 [ 0.2296679 ]
 [-0.28315711]
 [ 0.18720484]]

Best,
Achintya

**Comment by achintyasarkar** [comment]
it has been rectified.
Problem was due to not importing from __future__ import division
